### ISSUE TITLE
Incorrect links generated for cross-references with same name when using tagfiles

### ISSUE THREAD

**Issue by tttapa** [doxygen/doxygen#10850]
**Describe the bug**  
When an external project and the current project both have a MarkDown page with the same name/label, references in the current project resolve to the version of the page of the external project (instead of the current one).

**Screenshots**

![image](https://github.com/doxygen/doxygen/assets/21296729/a967d797-0e63-429a-b754-ffc7f7587e29)

The text refers to "Examples B" (which is the current project's version of the page), but the link points to the external project's version of the page (which has the title "Examples A").

![image](https://github.com/doxygen/doxygen/assets/21296729/816189e4-8fc8-480d-b62e-0e6a7689999b)

The "Related Pages" index is also incorrect (and doesn't even link to the current project's version of the examples page at all).

**To Reproduce**  
[Doxygen-10850.zip](https://github.com/doxygen/doxygen/files/15193092/Doxygen-10850.zip)

**Expected behavior**  
Inside of project B, the `@ref examples` command should link to project B's version of the example.md file.

Bonus: it would be nice to be able to explicitly refer to project A's version of the file as well.

**Version**  
1.10.0 (ebc57c6dd303a980bd19dd74b8b61c8f3f5180ca)

**Additional context**  
In my specific use case, I'm using the following in my `DoxygenLayout.xml`:

```xml
  <navindex>
    ...
    <tab type="user" visible="yes" title="Examples" url="@ref examples" intro=""/>
  </navindex>
```
The `@ref examples` produces the wrong link when using a tagfile from a project that also has a page labeled `examples`.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10854

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by tttapa** [comment]
Thanks a lot!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
